src/main/java/armazem/Armazem.java

```diff
@@ -0,0 +1,67 @@
+package armazem;
+
+import exceptions.IngredienteJaCadastradoException;
+import exceptions.IngredienteNaoEncontradoException;
+import exceptions.QuantidadeInvalidaException;
+import ingredientes.Ingrediente;
+
+import java.util.TreeMap;
+
+public class Armazem {
+
+    private final TreeMap<Ingrediente, Integer> estoque;
+
+    public Armazem() {
+        estoque = new TreeMap<>();
+    }
+
+    public void cadastrarIngredienteEmEstoque(Ingrediente ingrediente) {
+        if (!estoque.containsKey(ingrediente)) {
+            this.estoque.put(ingrediente, 0);
+        } else {
+            throw new IngredienteJaCadastradoException();
+        }
+    }
+
+    public Integer consultarQuantidadeDoIngredienteEmEstoque(Ingrediente ingrediente) {
+        checkIfIngredientExists(ingrediente);
+        return estoque.get(ingrediente);
+
+    }
+
+    public void descadastrarIngredienteEmEstoque(Ingrediente ingrediente) {
+        checkIfIngredientExists(ingrediente);
+        this.estoque.remove(ingrediente);
+    }
+
+    public void adicionarQuantidadeDoIngredienteEmEstoque(Ingrediente ingrediente, Integer quantity) {
+        checkIfQuantityIsvalid(quantity);
+        checkIfIngredientExists(ingrediente);
+        estoque.put(ingrediente, estoque.get(ingrediente) + quantity);
```

Comment:
Um outro método também que você poderia utilizar aqui é o computeIfPresent, ficaria assim a implementação:
`estoque.computeIfPresent(ingrediente, (key, value) -> value + quantity);`

Vou deixar um link aqui para você, caso tenha curiosidade
https://www.geeksforgeeks.org/hashmap-computeifpresentkey-bifunction-method-in-java-with-examples/